test/chpldoc/findUndocumentedSymbols/PREDIFF

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python3
+
+import sys
+import os
+import subprocess as sp
+import shlex
+
+testname = sys.argv[1]
+outfile = sys.argv[2]
+tmpfile = f"{outfile}.prediff.tmp"
+extra_args_ext = "finder-settings"
+extra_args_file = f"{testname}.{extra_args_ext}"
+
+chpl_home = os.getenv("CHPL_HOME", None)
+
+script = os.path.join(chpl_home, "tools", "chpldoc", "findUndocumentedSymbols")
+
+
+def get_prediff_opts():
+    if os.path.isfile(extra_args_file):
+        with open(extra_args_file, "r") as f:
+            lines = f.readlines()
+            if len(lines) == 1:
+                return shlex.split(lines[0].strip())
+            else:
+                print(f"Invalid '.{extra_args_ext}' file", sys.stderr)
+    return []
+
+
+def run_script(script, options, testfile, outfile):
+    args = [script] + options + [testfile]
+    with open(outfile, "w") as of:
+        cp = sp.run(args, stdout=of, stderr=of)
+        return cp.returncode == 0
+
+
+def append_to_file(file1, file2):
+    lines = []
```

Comment:
Python variables escape the scope of `with` blocks (or rather, `for`, `while`, etc. don't actually create new scopes...) so technically you don't need this `lines = []` assignment. I have no problem with keeping it anyway, though.

Reply:
That is something I actively avoid, I think it makes code hard to read